### DISCUSSION TITLE
How does Inflate work?

### DISCUSSION THREAD

**Discussion by mbaros100** [zxing-cpp/zxing-cpp#465] [Q&A]
I have generated a 1d UPC barcode without specifying the size (leaving width and height to 0) and the library generated a barcode of size 96x1

Then i tried to resize it to 210x70 by simply calling 
matrix = ZXing::Inflate(std::move(matrix), 210, 70, 1);

What it did, was actually keeping the barcode as a single line (2 lines to be precise) in the middle row and filled the rest of the data with white pixels?

So my question is: Does the Inflate function scale the barcode or not? If so, what am i doing wrong?
Thanks a lot.

EDIT:
For 2d barcodes it works fine but for 1d when i specify ZXing::Inflate(std::move(matrix), 200, 200, 0) it generates the following:
![a](https://user-images.githubusercontent.com/8537079/208523090-6ae247d7-10b9-4ca3-baaf-4188ebbe8407.png)

**Comment by axxel** [comment]
Inflate does not change the aspect ratio of the input. For 2D codes it would break the symbol in that case. It generally does not know what the matrix contains. I have a limited knowledge of the generator/writer side of the code base but I can imagine that all 1D writers output 1 pixel height images. Since there is no 2D symbol that is only 1pixel high I'd be happy to merge a change to Inflate that does something more sensible in that case, i.e. duplicating the rows to fill the image. Would you be willing to provide such a PR?

**Comment by axxel** [comment]
This is part of a now deprecated API.
